sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Processor/EventProcessorHost.cs

```diff
@@ -78,208 +79,41 @@ internal class Processor : EventProcessor<Partition>
-                    {
-                        // In this case, another host has claimed the partition before we did. That's safe to ignore. We'll still try to claim other partitions.
-                    }
-                }
-
-                return claimedOwnerships;
+                return await CheckpointStore.ClaimOwnershipAsync(desiredOwnership, cancellationToken).ConfigureAwait(false);
             }
 
             protected override async Task<IEnumerable<EventProcessorCheckpoint>> ListCheckpointsAsync(CancellationToken cancellationToken)
             {
-                // First, we read information from the location that the EventHubs V5 SDK writes to.
-                Dictionary<string, EventProcessorCheckpoint> checkpoints = new Dictionary<string, EventProcessorCheckpoint>();
-                string checkpointBlobsPrefix = string.Format(CultureInfo.InvariantCulture, CheckpointPrefixFormat, FullyQualifiedNamespace.ToLowerInvariant(), EventHubName.ToLowerInvariant(), ConsumerGroup.ToLowerInvariant());
-
-                await foreach (BlobItem item in ContainerClient.GetBlobsAsync(traits: BlobTraits.Metadata, prefix: checkpointBlobsPrefix, cancellationToken: cancellationToken).ConfigureAwait(false))
-                {
-                    if (long.TryParse(item.Metadata[OffsetMetadataKey], NumberStyles.Integer, CultureInfo.InvariantCulture, out long offset) &&
-                        long.TryParse(item.Metadata[SequenceNumberMetadataKey], NumberStyles.Integer, CultureInfo.InvariantCulture, out long sequenceNumber))
-                    {
-                        string partitionId = item.Name.Substring(checkpointBlobsPrefix.Length);
-
-                        LeaseInfos.TryAdd(partitionId, new LeaseInfo(offset, sequenceNumber));
-                        checkpoints.Add(partitionId, new EventProcessorCheckpoint()
-                        {
-                            ConsumerGroup = ConsumerGroup,
-                            EventHubName = EventHubName,
-                            FullyQualifiedNamespace = FullyQualifiedNamespace,
-                            PartitionId = partitionId,
-                            StartingPosition = EventPosition.FromOffset(offset, isInclusive: false)
-                        });
-                    }
-                }
-
-                // Check to see if there are any additional checkpoints in the older location that the V4 SDK would write to. If so, use them (this is helpful when moving from the V4 to V5 SDK,
-                // since it means we will not have to reprocess messages processed and checkpointed by the older SDK).
-                string legacyCheckpointAndOwnershipPrefix = $"{LegacyCheckpointStorageBlobPrefix}{ConsumerGroup}/";
-                await foreach (BlobItem item in ContainerClient.GetBlobsAsync(prefix: legacyCheckpointAndOwnershipPrefix, cancellationToken: cancellationToken).ConfigureAwait(false))
```

Comment:
I've removed the legacy compatibility mode because it wasn't tested and we are not sure exactly how we'd like to get it implemented. I'll file an issue to design and bring back the feature.